src/api/app/models/workflow/step.rb

```diff
@@ -64,14 +66,19 @@ def source_project_name
     step_instructions[:source_project]
   end
 
-  def target_package_name
+  def target_package_name(short_commit_sha: true)
     package_name = step_instructions[:target_package] || source_package_name
 
     case
     when scm_webhook.pull_request_event?
       package_name
     when scm_webhook.push_event?
-      "#{package_name}-#{scm_webhook.payload[:commit_sha]}"
+      commit_sha = scm_webhook.payload[:commit_sha]
+      if short_commit_sha
+        "#{package_name}-#{commit_sha[0..SHORT_COMMIT_SHA_LENGTH]}"
```

Comment:
With the following approach, we can set `SHORT_COMMIT_SHA_LENGTH` to `7`:

```suggestion
        "#{package_name}-#{commit_sha.slice(0, SHORT_COMMIT_SHA_LENGTH)}"
```